python.d/couchdb.chart.py

```diff
@@ -0,0 +1,411 @@
+        system = self.url + '/_node/{node}/_system'.format(node=self.node)
+        self.methods = [METHODS(get_data=self._get_overview_stats,
+                                url=stats,
+                                stats=OVERVIEW_STATS),
+                        METHODS(get_data=self._get_active_tasks_stats,
+                                url=active_tasks,
+                                stats=None),
+                        METHODS(get_data=self._get_overview_stats,
+                                url=system,
+                                stats=SYSTEM_STATS),
+                        METHODS(get_data=self._get_dbs_stats,
+                                url=self.url,
+                                stats=DB_STATS)]
+        # must initialise manager before using _get_raw_data
+        self._manager = self._build_manager()
+        self.dbs = [db for db in self.dbs
+                    if self._get_raw_data(self.url + '/' + db)]
+        for db in self.dbs:
+            self.definitions['db_sizes_file']['lines'].append(
+                ['db_'+db+'_sizes_file', db, 'absolute', 1, 1000]
+            )
+            self.definitions['db_sizes_external']['lines'].append(
+                ['db_'+db+'_sizes_external', db, 'absolute', 1, 1000]
+            )
+            self.definitions['db_sizes_active']['lines'].append(
+                ['db_'+db+'_sizes_active', db, 'absolute', 1, 1000]
+            )
+            self.definitions['db_doc_counts']['lines'].append(
+                ['db_'+db+'_doc_count', db, 'absolute']
+            )
+            self.definitions['db_doc_del_counts']['lines'].append(
+                ['db_'+db+'_doc_del_count', db, 'absolute']
+            )
+        return UrlService.check(self)
+
+    def _get_data(self):
+        threads = list()
+        queue = Queue()
+        result = dict()
+
+        for method in self.methods:
+            th = Thread(target=method.get_data,
+                        args=(queue, method.url, method.stats))
+            th.start()
+            threads.append(th)
+
+        for thread in threads:
+            thread.join()
+            result.update(queue.get())
+
+        # self.info('couchdb result = ' + str(result))
+        return result or None
+
+    def _get_overview_stats(self, queue, url, stats):
+        raw_data = self._get_raw_data(url)
+        if not raw_data:
+            return queue.put(dict())
+        data = loads(raw_data)
+        to_netdata = fetch_data(raw_data=data, metrics=stats)
+        if 'message_queues' in data:
+            to_netdata['peak_msg_queue'] = get_peak_msg_queue(data)
+        return queue.put(to_netdata)
+
+    def _get_active_tasks_stats(self, queue, url, _):
+        taskdict = defaultdict(int)
+        taskdict["activetasks_indexer"] = 0
+        taskdict["activetasks_database_compaction"] = 0
+        taskdict["activetasks_replication"] = 0
+        taskdict["activetasks_view_compaction"] = 0
```

Comment:
Why do we need to initialize these values? Can they not be in the `data`?

Reply:
Yes, sometimes they're not in the data; if there are no active tasks of that type, the entire stanza will be missing.